croncheck.php

```diff
@@ -234,7 +234,10 @@
     $delay .= "SCHEDULED TASK: " . get_class($task) . ' (' .$task->get_name() . ") Delay: $faildelay\n";
 }
 
+// Find any failed adhoc tasks.
 $records = $DB->get_records_sql('SELECT * from {task_adhoc} WHERE faildelay > 0');
+$adhoctaskdelays = [];
+
```

Comment:
This all looks fine but why not do it all in sql and group by with max(failday) ?